src/lib/detection_algorithms/cntkfastrcnn/exporter.js

```diff
@@ -0,0 +1,121 @@
+                function updateIndecies(testIndecies, cb) {
+                    self.posFrameIndex = 0;
+                    self.testFrameIndecies = testIndecies;
+                    cb();
+                },
+                async.each.bind(null,[self.exportDirPath, self.posDirPath, self.negDirPath, self.testDirPath], detectionUtils.ensureDirExists)
+            ], (err) => {
+                if (err) {
+                    reject(err);
+                    return;
+                }
+                resolve();
+            });
+        });
+    }
+
+    // Export a single frame to the training data
+    // Parameters:
+    //  frameFileName - The file name to use when saving the image file 
+    //  frameBuffer - A buffer with the frame image data 
+    //  bboxes  - a list of bboxes in the format of x1, y1, x2, y2 where the
+    //           coordinates are in absolute values of the image
+    //  tags - a list of objects containing the tagging data. Each object is in the format of:
+    //         {'x1' : int, 'y1' : int, 'x2' : int, 'y2' : int, 'class' : string}
+    //         Where (x1,y1) and (x2,y2) are the coordinates of the top left and bottom right corner
+    //         of the bounding boxes (respectively), and 'class' is the name of the class.
+    // Returns: A Promise object that resolves when the operation completes
+    this.exportFrame = function exportFrame(frameFileName, frameBuffer, tags) {
+        return new Promise((resolve, reject) => {
+            async.waterfall(
+              [
+                async.each.bind(null,[path.join(self.posDirPath,  frameFileName), 
+                                      path.join(self.negDirPath,  frameFileName), 
+                                      path.join(self.testDirPath, frameFileName),
+                                      path.join(self.posDirPath,  `${frameFileName}.bboxes.tsv`),
+                                      path.join(self.testDirPath, `${frameFileName}.bboxes.tsv`),
+                                      path.join(self.posDirPath,  `${frameFileName}.bboxes.labels.tsv`), 
+                                      path.join(self.testDirPath, `${frameFileName}.bboxes.labels.tsv`)], detectionUtils.deleteFileIfExists),
+
+                function determineWritePath(cb){
+                    if( !tags.length){ 
+                        cb(null, self.negDirPath);
+                    } else {
+                        var dirPath = self.testFrameIndecies.includes(self.posFrameIndex) ? self.testDirPath : self.posDirPath;
+                        self.posFrameIndex++;
+                        cb(null, dirPath);
+                    }
+                },
+                function saveImage(exportPath, cb) {
+                    var imageFilePath  = path.join(exportPath, frameFileName);
+                    fs.writeFile(imageFilePath, frameBuffer, () => {
+                        cb(null, exportPath);
```

Comment:
Handle error in case the write fails, e.g. (get the error object in the callback and pass it to cb)